huaweicloud/services/tms/resource_huaweicloud_tms_resource_tags.go

```diff
@@ -0,0 +1,274 @@
+	}
+
+	return resourceResourceTagsRead(ctx, d, meta)
+}
+
+func FlattenTagsToMap(tagsResp []tags.ResourceTag) map[string]interface{} {
+	result := make(map[string]interface{})
+	for _, val := range tagsResp {
+		result[val.Key] = val.Value
+	}
+	return result
+}
+
+func compareTwoTags(localTags, remoteTags map[string]interface{}) (same, diff map[string]interface{}) {
+	same = make(map[string]interface{})
+	diff = make(map[string]interface{})
+
+	for localKey, localVal := range localTags {
+		if remoteVal, ok := remoteTags[localKey]; ok {
+			local, isTypeLocalOk := localVal.(string)
+			if !isTypeLocalOk {
+				log.Printf("The type of tag key in the script is incorrect, want 'string', but got '%T'", localVal)
+			}
+			remote, isTypeRemoteOk := remoteVal.(*string)
+			if !isTypeRemoteOk {
+				log.Printf("The type of tag key in the remote response is incorrect, want '*string', but got '%T'", remoteVal)
+			}
+			if local == *remote {
+				same[localKey] = localVal
+				continue
+			}
+		}
+		diff[localKey] = localVal
+	}
+	return
+}
+
+func resourceResourceTagsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(*config.Config).TmsV2Client()
+	if err != nil {
+		return diag.Errorf("error creating TMS v2 client: %s", err)
+	}
+
+	var (
+		projectId = d.Get("project_id").(string)
+		resources = d.Get("resources").([]interface{})
+		resResult = make([]interface{}, 0, len(resources))
+		tagsInput = d.Get("tags").(map[string]interface{})
+
+		isTagsConfigured = false
+	)
+
+	// Check whether all tagged resources contain the expected tags correctly. If not, inconsistent tags information
+	// will be printed in the log.
+	for _, val := range resources {
+		resource := val.(map[string]interface{})
+		resourceId := resource["resource_id"].(string)
+		opts := tags.QueryOpts{
+			ResourceId:   resourceId,
+			ResourceType: resource["resource_type"].(string),
+			ProjectId:    projectId,
+		}
+		resp, err := tags.Get(client, opts)
+		if err != nil {
+			return diag.Errorf("error query resource (%s) tags: %s", resourceId, err)
+		}
+		actualTags := FlattenTagsToMap(resp)
+		same, diff := compareTwoTags(tagsInput, actualTags)
+		if len(diff) > 0 {
+			log.Printf("[ERROR] The tags of resource (%s) don't contain some tags that are expected to need to be set."+
+				" It should contain tags (%#v), but some tags were not set successfully: %#v", resourceId, tagsInput,
+				actualTags)
+		}
+		// If the tags are queried from the resource side (even only part of tags are queried), it means that the
+		// creation or update action of the resource is (partially) successful.
+		if len(same) > 0 {
+			if !isTagsConfigured {
+				isTagsConfigured = true
+			}
+			resResult = append(resResult, val)
+		}
+	}
+	if !isTagsConfigured {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "TMS tags management")
+	}
+	mErr := multierror.Append(nil,
+		d.Set("resources", resResult),
+		d.Set("tags", tagsInput),
```

Comment:
the `tagsInput` get from the schema, do not need to set it again.

Reply:
okey